spec/jekyll-linkpreview_spec.rb

```diff
@@ -521,55 +523,62 @@ def template_dir
     end
   end
 
-  describe '#render' do
-    before do
-      @title = 'awesome.org - an awesome organization in the world'
-      @domain = 'awesome.org'
-      @url = "https://#{@domain}/about"
-      @image = "https://#{@domain}/images/favicon.ico"
-      @description = 'An awesome organization in the world.'
-      tokenizer = Liquid::Tokenizer.new('')
-      parse_context = Liquid::ParseContext.new
-      @tag = TestLinkpreviewTag.parse(nil, @url, tokenizer, parse_context)
-
-      Dir.mkdir @tag.template_dir
-    end
+  where(:site_source) do
+    [
+      [Dir.pwd], # Default value for Jekyll
+      ["."], # Explicitly specified
+      ["_content"], # Modified
+    ]
+  end
+  with_them do
+    source = params[:site_source]
+    describe '#render' do
+      before do
+        @title = 'awesome.org - an awesome organization in the world'
+        @domain = 'awesome.org'
+        @url = "https://#{@domain}/about"
+        @image = "https://#{@domain}/images/favicon.ico"
+        @description = 'An awesome organization in the world.'
+        tokenizer = Liquid::Tokenizer.new('')
+        parse_context = Liquid::ParseContext.new
+        @tag = TestLinkpreviewTag.parse(nil, @url, tokenizer, parse_context)
+        Dir.mkdir File.join(source, @tag.template_dir)
```

Comment:
```suggestion
        FileUtils.mkdir_p File.join(source, @tag.template_dir)
```

Without this change, `rake test` would throw an error like this because `_content` directory does not exist. We would need to run the equivalent of `mkdir -p`.

```
  1) Jekyll::Linkpreview::LinkpreviewTag site_source: "_content" #render custom template for OpenGraphProperties when a custom template file for OpenGraphProperties exists can render custom template
     Got 0 failures and 2 other errors:

     1.1) Failure/Error: Dir.mkdir File.join(source, @tag.template_dir)
          
          Errno::ENOENT:
            No such file or directory @ dir_s_mkdir - _content/_includes
          # ./spec/jekyll-linkpreview_spec.rb:545:in `mkdir'
          # ./spec/jekyll-linkpreview_spec.rb:545:in `block (4 levels) in <top (required)>'

     1.2) Failure/Error: FileUtils.rm_r File.join(source, @tag.template_dir)
          
          Errno::ENOENT:
            No such file or directory @ apply2files - _content/_includes
          # ./spec/jekyll-linkpreview_spec.rb:549:in `block (4 levels) in <top (required)>'
```

Reply:
Whoops, good catch! I didn't notice it because every test leaves behind the `_content/.jekyll-cache` directory.

The suggested change has been committed as-is via GitHub. I didn't think GitHub would auto-resolve this discussion when I did that, though!